builder/builder.go

```diff
@@ -10,19 +10,22 @@ import (
 	bfttypes "github.com/cometbft/cometbft/types"
 	"github.com/ethereum-optimism/optimism/op-service/eth"
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/trie"
 	"github.com/polymerdao/monomer"
 	"github.com/polymerdao/monomer/app/peptide/store"
 	"github.com/polymerdao/monomer/app/peptide/txstore"
 	"github.com/polymerdao/monomer/mempool"
 )
 
 type Builder struct {
-	mempool    *mempool.Pool
-	app        monomer.Application
-	blockStore store.BlockStore
-	txStore    txstore.TxStore
-	eventBus   *bfttypes.EventBus
-	chainID    monomer.ChainID
+	mempool      *mempool.Pool
+	app          monomer.Application
+	blockStore   store.BlockStore
+	txStore      txstore.TxStore
+	eventBus     *bfttypes.EventBus
+	chainID      monomer.ChainID
+	ethStateTrie *trie.StateTrie
```

Comment:
It's a bit confusing (I'd like to change it in the future), but the builder controls rollbacks. So we need to rollback the state trie in `Rollback`

Reply:
Similar: `get_Proof` api wants to be callable by block hash / number. As far as I can tell, this implementation doesn't store historical state trie data, so those get proof calls can't be computed other than at the current block with the in-memory trie.

Reply:
@NiloCK Yeah that's a good point. I'm currently working on making the DB snapshots indexable by block root to support rollbacks so as long as we have the block hash/number we should be able to get the block root off of the header to use for `get_proof`.